lualib/rspamadm/ready.lua

```diff
@@ -0,0 +1,164 @@
+      :description 'Polling interval in seconds'
+      :argname('interval')
+      :default('1')
+parser:flag '--no-ssl-verify'
+      :description 'Disable SSL verification'
+      :argname('no_ssl_verify')
+parser:flag '-q --quiet'
+      :description 'Only output errors'
+      :argname('quiet')
+parser:flag '-v --verbose'
+      :description 'Output more information'
+      :argname('verbose')
+
+local http_params = {
+  config = rspamd_config,
+  ev_base = rspamadm_ev_base,
+  session = rspamadm_session,
+  resolver = rspamadm_dns_resolver,
+}
+
+local function load_config(config_file)
+  local _r, err = rspamd_config:load_ucl(config_file)
+
+  if not _r then
+    rspamd_logger.errx('cannot load %s: %s', config_file, err)
+    os.exit(1)
+  end
+
+  _r, err = rspamd_config:parse_rcl({ 'logging', 'worker' })
+  if not _r then
+    rspamd_logger.errx('cannot process %s: %s', config_file, err)
+    os.exit(1)
+  end
+
+  if not rspamd_config:init_modules() then
+    rspamd_logger.errx('cannot init modules when parsing %s', config_file)
+    os.exit(1)
+  end
+
+  rspamd_config:init_subsystem('symcache')
+end
+
+local function poll_ready(args)
+  local total_timeout = tonumber(args.timeout)
+  local interval = tonumber(args.interval)
+  local url = args.url
+  
+  -- Add /ready endpoint if not specified
+  if not url:match("/ready$") then
+    url = lua_util.remove_trailing_slash(url) .. "/ready"
+  end
+  
+  local start_time = os.time()
+  local attempts = 0
+  local exit_code = 1
+  
+  while os.time() - start_time < total_timeout do
+    attempts = attempts + 1
+    
+    if not args.quiet then
+      io.stdout:write(string.format("Attempt %d: Checking if Rspamd is ready...\n", attempts))
+    end
+    
+    local err, response = rspamd_http.request({
+      url = url,
+      config = rspamd_config,
+      ev_base = rspamadm_ev_base,
+      session = rspamadm_session,
+      resolver = rspamadm_dns_resolver,
+      log_obj = rspamd_config,
+      no_ssl_verify = args.no_ssl_verify,
+    })
+    
+    if err then
+      if args.verbose then
+        io.stderr:write(string.format("Error checking Rspamd status: %s\n", err))
+      end
+    elseif response and response.code == 200 then
+      if not args.quiet then
+        io.stdout:write("Rspamd is ready and operational!\n")
+      end
+      exit_code = 0
+      break
+    else
+      local status_code = response and response.code or "unknown"
+      if args.verbose then
+        io.stderr:write(string.format("Rspamd not ready (status code: %s)\n", status_code))
+      end
+    end
+    
+    if os.time() - start_time + interval >= total_timeout then
+      if not args.quiet then
+        io.stderr:write("Timeout reached. Rspamd is not ready.\n")
+      end
+      break
+    end
+    
+    if not args.quiet and args.verbose then
+      io.stdout:write(string.format("Waiting %d seconds before next attempt...\n", interval))
+    end
+    
+    lua_util.sleep(interval)
```

Comment:
Method doesn't exist; we might need to reach for FFI here

Reply:
We can use `rspamadm_ev_base` to add async timers.